[kerktaiheng] iP

kerktaiheng opened the pull request:
# alice – tsundere productivity chatbot😒

> “shut up nerd” –alice [source]

Alice, the bot that hates you, is here to help you free your mind of having to remember things you need to do. It’s:

- **Command-based** (because in her eyes you can't handle more than that)
- **Easy to learn** (but literally impossible to master)
- **Incredibly motivational** (In an optimistic sense)

Usage guide:

1. Don't

Ok, for real

1. `git clone`
2. `gradle build`
3. **Add your tasks** using the commands, which she won't be helpful with to be honest

## Features:
- Managing todos, events, and deadlines
- Find stuff with fuzzy search
- Get "motivated" when you make mistakes and fail to achieve your goals

[mandatory link](https://ozpropets.com/wp-content/uploads/2022/01/5-Signs-Your-Cat-Hates-YOU_00-jpg.webp)

mandatory task list in alice's pov
- [X] go away
- [ ] come back 

mandatory codeblock
```java
public class Main {
    public static void main(String[] args) {
        goAway();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kerktaiheng committed:
Rename chatbot to Alice, greet and exit

kerktaiheng committed:
Echo user commands

kerktaiheng committed:
List commands

kerktaiheng committed:
Mark and unmark tasks for completion

kerktaiheng committed:
Support for ToDos, Events, Deadlines

kerktaiheng committed:
Automated UI Testing

kerktaiheng committed:
Some error handling

kerktaiheng committed:
Delete tasks

kerktaiheng committed:
Add MessageGenerator for personality

kerktaiheng committed:
Add Exceptions

kerktaiheng committed:
Add Task Saving and Loading

kerktaiheng committed:
Merge branch 'branch-Level-7'

kerktaiheng committed:
Add DateTime support

kerktaiheng committed:
Merge branch 'branch-Level-8'

kerktaiheng committed:
Add More OOP, packages, Collections

kerktaiheng committed:
Merge remote-tracking branch 'origin/add-gradle-support'

kerktaiheng committed:
Add Gradle support

kerktaiheng committed:
Add JUnit Tests

kerktaiheng committed:
Add JavaDoc and Google Java codestyle

kerktaiheng committed:
Merge branch 'branch-Javadoc-CodingStandard'

kerktaiheng committed:
Add find

kerktaiheng committed:
Merge branch 'branch-Level-9'

kerktaiheng committed:
Correct coding style

Ryuse commented on src/main/java/command/Parser.java:
```diff
@@ -0,0 +1,129 @@
+
+public class Parser {
+
+    private static final DateTimeFormatter DATETIMEFORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
Perhaps use `DATETIME_FORMATTER` instead such that it follows the coding standard of screaming camel case?

kea-S replied:
I agree with @Ryuse, it should be DATETIME_FORMATTER or maybe even DATE_TIME_FORMATTER

Ryuse commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
Perhaps a more specific name might be better instead of just `FORMATTER`? 
Just in case different formatters are needed in the future.

kerktaiheng replied:
yeah ure right

Ryuse commented on src/main/java/ui/MessageGenerator.java:
```diff
@@ -0,0 +1,284 @@
+        "i hope i never have to deal with you again",
+        "never return",};
+
+    private static final String[] GENERICINSULTS = {
```
Maybe an underline inbetween `GENERICINSULTS` may be better for readability? I noticed the same issue in several other places too.

Ryuse commented on src/main/java/exceptions/AliceException.java:
```diff
@@ -0,0 +1,14 @@
+     * Constructs a new AliceException with no detail message.
+     */
+    public AliceException() {
+        super("");
```
May I know why the exception message is empty? Is it intended to be used with `MessageGenerator.java`?

Ryuse commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,136 @@
+     * Displays an error message.
+     */
+    public void showError() {
+        System.out.println(box(MessageGenerator.getErrorMsg()));
```
Would a more descriptive error message may be better for the user?

kerktaiheng replied:
i tried telling her, she straight up just alt-f4'd

Ryuse commented on src/main/java/ui/MessageGenerator.java:
```diff
@@ -0,0 +1,284 @@
+
+import utils.RandomUtils;
+
+public class MessageGenerator {
```
I like the idea of the random message generator. It adds more personality to Alice.

kerktaiheng replied:
she hates me tho

Ryuse replied:
But with more options :D

Ryuse commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,23 @@
+/**
+ * Represents a command that can be executed.
+ */
+public abstract class Command {
```
I like the usage of the `Command` class and its subclasses. Makes the code cleaner and more scalable for the future.

Ryuse commented on src/main/java/data/alice.txt:
```diff
@@ -0,0 +1 @@
+T|0|kill myself
```
Perhaps adding `alice.txt` to `.gitignore` would be good? This is so when the repo is cloned the data won't be cloned too. Also, are you good?

kerktaiheng replied:
yeah ure right im trolling

Ryuse reviewed (commented):
Your code was well written. There are some minor issues with the coding standard but overall it looks good.

Ryuse commented on src/main/java/command/FindCommand.java:
```diff
@@ -0,0 +1,30 @@
+
+    private String keyword;
+
+    public FindCommand(String keyword) {
```
Maybe some java docs here?

Ryuse reviewed (commented):
I like the code quality and it seems to be mostly compliant with the coding standard.

kea-S commented on src/main/java/Alice.java:
```diff
@@ -0,0 +1,64 @@
+
```
Just a nit! But there shouldn't be an empty line at the start of the file.

kerktaiheng replied:
yeah ure right

kea-S commented on src/main/java/Alice.java:
```diff
@@ -0,0 +1,64 @@
+        }
+    }
+
+    /**
```
Loving the javadocs comments. They are very descriptive

kerktaiheng replied:
lol

kea-S commented on src/main/java/command/EventCommand.java:
```diff
@@ -0,0 +1,47 @@
+        this.to = to;
+    }
+
+    /**
```
Just a thought but maybe you can link each word referencing a class you created to the class in the javadocs, or maybe use the see also annotation

kerktaiheng replied:
actually what does this mean

kea-S reviewed (commented):
Overall very clean code! Good job.

kerktaiheng committed:
Add Varargs to Task constructor

kerktaiheng committed:
Add JavaFX

kerktaiheng committed:
Merge branch 'branch-Level-10'

kerktaiheng committed:
Remove Text UI Testing

kerktaiheng committed:
Misc fixes

kerktaiheng committed:
Fix UI

The UI is fixed to resemble the minimal requirements of a chat-like
application.

In particular, resizing and dialog box alignment should no longer break
the apperance of the app.

kerktaiheng committed:
Add Assertions

kerktaiheng committed:
Refactor

kerktaiheng committed:
Refactor Parser

kerktaiheng committed:
Merge pull request #1 from kerktaiheng/branch-A-Assertions

Add Assertions

kerktaiheng committed:
Merge branch 'master' of https://github.com/kerktaiheng/ip

kerktaiheng committed:
Merge branch 'master' into branch-A-CodeQuality

kerktaiheng committed:
Merge pull request #2 from kerktaiheng/branch-A-CodeQuality

Branch a code quality

kerktaiheng committed:
Add better GUI

kerktaiheng committed:
Add tests for Parser

kerktaiheng committed:
Add more error handling and error messages

kerktaiheng committed:
Add Javadoc

kerktaiheng committed:
Update markdowns

kerktaiheng committed:
Remove println

kerktaiheng committed:
Update README.md

kerktaiheng committed:
Remove println
